### DISCUSSION TITLE
Support for structural search and replace

### DISCUSSION THREAD

**Discussion by irevoire** [helix-editor/helix#11366] [Q&A]
Hello, I just discovered the structural search and replace which would greatly help me right now, but it doesn’t seem to be supported by helix. Am I right?

I tried to put my cursor on the line in multiple places, select the line, etc, and run the code action, but it doesn’t recognize the pattern.

https://rust-analyzer.github.io/manual.html#structural-search-and-replace

**Comment by the-mikedavis** [comment]
This is supported actually since rust-analyzer exposes it through code actions. If you type a valid SSR in a comment and then use `<space>a` on that line you should see "Apply SSR in file" and "Apply SSR in workspace"

**Reply by irevoire** [reply]
That's what I initially thought, but it doesn't work on my side.
I literally copy pasted their example and wrote one that should work in my small file. And this is the output of `<space>a` whatever is the position of my pointer:
<img width="1014" alt="image" src="https://github.com/user-attachments/assets/57b32690-fc8b-4f7e-86c5-5dc9fac123cb">


Also, here's some info about my helix config:
<img width="951" alt="image" src="https://github.com/user-attachments/assets/cb39f90d-4b02-40cb-851f-94877ea64833">

And FYI, `<space>a` works for other commands like renaming a variable or importing stuff.

**Reply by the-mikedavis** [reply]
I think there might be an issue with how rust-analyzer tries to parse SSRs from comments. I can reproduce that example not working while with some edits this does work for me:

```rust
fn main() {
    let ret = do_add(1, 2);
    println!("Hello, world! {ret:?}");
}

fn do_add(a: u32, b: u32) -> u32 {
    u32::checked_add(a, b)
}

// do_add($a, $b) ==>> ($a).checked_add($b)
```
